microsoft-edge/progressive-web-apps-chromium/how-to/digital-goods-api.md

```diff
@@ -0,0 +1,217 @@
+
+To mark a purchase as "consumed", use the `consume` method:
+
+```javascript
+digitalGoodsService.consume(purchaseToken);
+```
+
+
+<!-- ====================================================================== -->
+## Checking existing purchases (listPurchases)
+
+The `listPurchases` method returns information about the user's existing purchases.  This method allows a client to get a list of items that are currently owned or purchased by the user.  This may be necessary, to do either of the following:
+
+* Check for entitlements, such as whether a subscription, promotional code, or permanent upgrade is active.
+
+* Recover from network interruptions during a purchase, such as when the item is purchased but not yet acknowledged.
+
+The `listPurchases` method returns item IDs and purchase tokens.  Before you grant an entitlement, you should verify the returned item ID or the returned purchase token, by using a direct developer-to-provider API, as shown below:
+
+```javascript
+purchases = await digitalGoodsService.listPurchases();
+for (p of purchases) {
+  VerifyAndGrantEntitlement(p.itemId, p.purchaseToken);
+}
+```
+
+The `listPurchases` method doesn't return consumed products or expired subscriptions.
+
+
+<!-- ====================================================================== -->
+## Purchase history (listPurchaseHistory)
+
+The `listPurchaseHistory()` method returns the most recent purchase made by the user for each item, regardless of whether the purchase is expired, canceled, or consumed.  This method returns a list of `PurchaseDetails` containing the `itemId` and `purchaseToken` for each purchase.
+
+```javascript
+purchases = await digitalGoodsService.listPurchaseHistory();
+for (p of purchases) {
+  VerifyAndCheckExpiredEntitlement(p.itemId, p.purchaseToken);
+}
+```
+
+
+<!-- ====================================================================== -->
+## See also
+
+* [Digital Goods API For Microsoft Store PWA Explainer](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/PwaDigitalGoods/explainer.md)
+* [StoreProduct.InAppOfferToken Property](/uwp/api/windows.services.store.storeproduct.inappoffertoken)
+
+W3C:
+* [Digital Goods API: Draft Community Group Report](https://wicg.github.io/digital-goods/)
+* [Payment Request API](https://www.w3.org/TR/payment-request/)
+* [5. PaymentCurrencyAmount dictionary](https://www.w3.org/TR/payment-request/#dom-paymentcurrencyamount) in _Payment Request API_.
+
+MDN:
+* [Intl.NumberFormat](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat)
+* [display_override](https://developer.mozilla.org/docs/Web/Manifest/display_override)<!-- todo: delete link? "override" not in article body   -->
```

Comment:
delete "display_override" Ref link?  "override" not in article body
https://developer.mozilla.org/docs/Web/Manifest/display_override
@runyuany